service/dynamodb/internal/customizations/accept_encoding_gzip.go

```diff
@@ -0,0 +1,168 @@
+// disable the underlying http client automatically enabling for gzip
+// decompress content-encoding support.
+type DisableAcceptEncodingGzipMiddleware struct{}
+
+// ID returns the id for the middleware.
+func (*DisableAcceptEncodingGzipMiddleware) ID() string {
+	return "DynamoDB:DisableAcceptEncodingGzipMiddleware"
+}
+
+// HandleFinalize implements the FinalizeMiddlware interface.
+func (*DisableAcceptEncodingGzipMiddleware) HandleFinalize(
+	ctx context.Context, input middleware.FinalizeInput, next middleware.FinalizeHandler,
+) (
+	output middleware.FinalizeOutput, metadata middleware.Metadata, err error,
+) {
+	req, ok := input.Request.(*smithyhttp.Request)
+	if !ok {
+		return output, metadata, &smithy.SerializationError{
+			Err: fmt.Errorf("unknown request type %T", input.Request),
+		}
+	}
+
+	// Explicitly enable gzip support, this will prevent the http client from
+	// auto extracting the zipped content.
+	req.Header.Set(acceptEncodingHeaderKey, "identity")
+
+	return next.HandleFinalize(ctx, input)
+}
+
+// AcceptEncodingGzipMiddleware provides a middleware to enable support for
+// gzip responses, with manual decompression. This prevents the underlying HTTP
+// client from performing the gzip decompression automatically.
+type AcceptEncodingGzipMiddleware struct{}
+
+// ID returns the id for the middleware.
+func (*AcceptEncodingGzipMiddleware) ID() string { return "DynamoDB:AcceptEncodingGzipMiddleware" }
+
+// HandleFinalize implements the FinalizeMiddlware interface.
+func (*AcceptEncodingGzipMiddleware) HandleFinalize(
+	ctx context.Context, input middleware.FinalizeInput, next middleware.FinalizeHandler,
+) (
+	output middleware.FinalizeOutput, metadata middleware.Metadata, err error,
+) {
+	req, ok := input.Request.(*smithyhttp.Request)
+	if !ok {
+		return output, metadata, &smithy.SerializationError{
+			Err: fmt.Errorf("unknown request type %T", input.Request),
+		}
+	}
+
+	// Explicitly enable gzip support, this will prevent the http client from
+	// auto extracting the zipped content.
+	req.Header.Set(acceptEncodingHeaderKey, "gzip")
+
+	return next.HandleFinalize(ctx, input)
+}
+
+// DecompressGzipMiddleware provides the middleware for decompressing a gzip
+// response from the service.
+type DecompressGzipMiddleware struct{}
+
+// ID returns the id for the middleware.
+func (*DecompressGzipMiddleware) ID() string { return "DynamoDB:DecompressGzipMiddleware" }
+
+// HandleDeserialize implements the DeserializeMiddlware interface.
+func (*DecompressGzipMiddleware) HandleDeserialize(
+	ctx context.Context, input middleware.DeserializeInput, next middleware.DeserializeHandler,
+) (
+	output middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	output, metadata, err = next.HandleDeserialize(ctx, input)
+	if err != nil {
+		return output, metadata, err
+	}
+
+	resp, ok := output.RawResponse.(*smithyhttp.Response)
+	if !ok {
+		return output, metadata, &smithy.SerializationError{
```

Comment:
I missed to point this before. This should be DeserializationError too

Reply:
thanks, i thought i fixed all of these, will go back and make sure.